openforcefield/topology/molecule.py

```diff
@@ -1913,24 +1916,68 @@ def generate_conformers(self,
-            return toolkit_registry.call('generate_conformers', self,
-                                         clear_existing)
+            return toolkit_registry.call('generate_conformers', self, num_conformers=num_conformers,
+                                         clear_existing=clear_existing)
         elif isinstance(toolkit_registry, ToolkitWrapper):
             toolkit = toolkit_registry
-            return toolkit.generate_conformers(self, clear_existing)
+            return toolkit.generate_conformers(self, num_conformers=num_conformers, clear_existing=clear_existing)
         else:
             raise InvalidToolkitError(
                 'Invalid toolkit_registry passed to generate_conformers. Expected ToolkitRegistry or ToolkitWrapper. Got  {}'
                 .format(type(toolkit_registry)))
 
+    def compute_partial_charges_am1bcc(self, toolkit_registry=GLOBAL_TOOLKIT_REGISTRY):
+        """
+        Calculate partial atomic charges for this molecule using an underlying toolkit
+
+        Parameters
+        ----------
+        toolkit_registry : openforcefield.utils.toolkits.ToolRegistry or openforcefield.utils.toolkits.ToolkitWrapper, optional, default=None
+            :class:`ToolkitRegistry` or :class:`ToolkitWrapper` to use for SMILES-to-molecule conversion
+
+        Examples
+        --------
+
+        >>> molecule = Molecule.from_smiles('CCCCCC')
+        >>> molecule.generate_conformers()
+        >>> molecule.compute_partial_charges()
+
+        Raises
+        ------
+        InvalidToolkitError
+            If an invalid object is passed as the toolkit_registry parameter
+
+        """
+        if isinstance(toolkit_registry, ToolkitRegistry):
+            charges = toolkit_registry.call(
+                      'compute_partial_charges_am1bcc',
+                      self
+            )
+        elif isinstance(toolkit_registry, ToolkitWrapper):
+            toolkit = toolkit_registry
+            charges = toolkit.compute_partial_charges_am1bcc(
+                self
+            )
+        else:
+            raise InvalidToolkitError(
+                'Invalid toolkit_registry passed to compute_partial_charges_am1bcc. Expected ToolkitRegistry or ToolkitWrapper. Got  {}'
+                .format(type(toolkit_registry)))
+        self.partial_charges = charges
+
+
     def compute_partial_charges(self,
-                                charge_model=None,
+                                #quantum_chemical_method='AM1-BCC',
+                                #partial_charge_method='None',
                                 toolkit_registry=GLOBAL_TOOLKIT_REGISTRY):
         """
         Calculate partial atomic charges for this molecule using an underlying toolkit
 
         Parameters
         ----------
+        quantum_chemical_method : string, default='AM1-BCC'
+            The quantum chemical method to use for partial charge calculation.
+        partial_charge_method : string, default='None'
```

Comment:
Since the whole compute_partial_charges stack is disabled, I left it in so there's a starting point for when we do re-implement it. I did some related tidying up in 4a77465